packages/react-scripts/config/env.js

```diff
@@ -19,21 +19,27 @@ function getClientEnvironment(publicUrl) {
     .keys(process.env)
     .filter(key => REACT_APP.test(key))
     .reduce((env, key) => {
-      env[key] = JSON.stringify(process.env[key]);
+      env[key] = process.env[key];
       return env;
     }, {
       // Useful for determining whether we’re running in production mode.
       // Most importantly, it switches React into the correct mode.
-      'NODE_ENV': JSON.stringify(
-        process.env.NODE_ENV || 'development'
-      ),
+      'NODE_ENV': process.env.NODE_ENV || 'development',
       // Useful for resolving the correct path to static assets in `public`.
       // For example, <img src={process.env.PUBLIC_URL + '/img/logo.png'} />.
       // This should only be used as an escape hatch. Normally you would put
       // images into the `src` and `import` them in code to get their paths.
-      'PUBLIC_URL': JSON.stringify(publicUrl)
+      'PUBLIC_URL': publicUrl
     });
-  return {'process.env': processEnv};
+
+  processEnv['process.env'] = Object
```

Comment:
As @gaearon [explanation](https://github.com/facebookincubator/create-react-app/pull/1440#issuecomment-278508740), We need to stringify values into strings such that we could use `process.env.X` in js code.

I have [changed](https://github.com/facebookincubator/create-react-app/pull/1440/commits/5763d44d9971ebbfc896052fbc7597c02e9feb34) some code for more explicitly.